### PR TITLE
fix(core/data-cite|xref): use xref hrefs when available

### PR DESCRIPTION
**Pull request by marcoscaceres**
Somewhat high priority... this is what was causing those weird URL issues that we saw the other day where, for example, we ended up with urls like TR/foo/TR/foo#whatever.

### CODE REVIEW

File: src/core/data-cite.js [hunk 1]
```diff
@@ -44,13 +44,18 @@ async function getLinkProps(citeDetails) {
...
-  if (frag) {
-    href = new URL(frag, href).href;
+  if (canonicalHref) {
+    // Xref gave us a canonical link, so let's use that.
+    href = canonicalHref;
+  } else {
+    if (path) {
+      // See: https://github.com/w3c/respec/issues/1856#issuecomment-429579475
+      const relPath = path.startsWith("/") ? `.${path}` : path;
+      href = new URL(relPath, href).href;
```
Comment (marcoscaceres) [inline src/core/data-cite.js:54]: This legacy garbage here was causing the bug :(

File: src/core/data-cite.js [hunk 2]
```diff
@@ -124,13 +127,14 @@ const findPath = makeComponentFinder("/");
  * @property {boolean} isNormative
  * @property {string} frag
  * @property {string} path
- *
+ * @property {string} href - canonical href coming from xref
```
Comment (sidvishnoi) [inline src/core/data-cite.js:130]:
```suggestion
 * @property {string} [href] - canonical href coming from xref
```

File: src/core/data-cite.js [hunk 3]
```diff
@@ -66,42 +71,40 @@ function linkElem(elem, linkProps, citeDetails) {
...
+      elem.textContent = title;
     }
-    anchor.href = href;
+    elem.href = href;
     if (wrapInCiteEl) {
       const cite = document.createElement("cite");
-      anchor.replaceWith(cite);
-      cite.append(anchor);
+      elem.replaceWith(cite);
+      cite.append(elem);
```
Comment (sidvishnoi) [inline src/core/data-cite.js:82]: Note for future: it's a refactor. No other changes.

File: src/core/data-cite.js [hunk 4]
```diff
@@ -66,42 +71,40 @@ function linkElem(elem, linkProps, citeDetails) {
...
+    elem.append(cite);
+  }
+  if ("export" in elem.dataset) {
+    const msg = "Exporting an linked external definition is not allowed.";
+    const hint = "Please remove the `data-export` attribute.";
+    showError(msg, name, { hint, elements: [elem] });
+    delete elem.dataset.export;
+  }
+  elem.classList.add("externalDFN");
+  elem.dataset.noExport = "";
```
Comment (sidvishnoi) [inline src/core/data-cite.js:107]: Note for future: it's a refactor. No other changes.

**Review by sidvishnoi — APPROVED** [review]

**Comment by sidvishnoi** [comment]
I'm gonna send a separate PR for the refactors in this PR.

**Review by sidvishnoi — APPROVED** [review]

**Pull request merged**